From 999f620944d8b1a7b350912c40e7b8e0ce7c272d Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Tue, 7 Nov 2023 16:33:07 +0100
Subject: [PATCH 0032/1695] plat-stm32mp2: activate
 CFG_CORE_HALT_CORES_ON_PANIC

Default halt other cores when panicking on STM32MP25x platforms. SGI15
is used for this purpose.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 9cc4faede28b2f39be1e3b405af2d2efe5ff1672 plat-stm32mp2: activate CFG_CORE_HALT_CORES_ON_PANIC]
Change-Id: I43687c05c6de3b95154c5dd6ea9b0a63cdcccb22
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
[etienne: cherry-picked from d53a711bdd0674a604d46ac6af67d3651644e214]
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
---
 core/arch/arm/plat-stm32mp2/conf.mk | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp2/conf.mk b/core/arch/arm/plat-stm32mp2/conf.mk
index 3cc30fce7..76dda5797 100644
--- a/core/arch/arm/plat-stm32mp2/conf.mk
+++ b/core/arch/arm/plat-stm32mp2/conf.mk
@@ -28,6 +28,7 @@ $(call force,CFG_DRIVERS_GPIO,y)
 $(call force,CFG_DRIVERS_PINCTRL,y)
 $(call force,CFG_DT,y)
 $(call force,CFG_GIC,y)
+$(call force,CFG_HALT_CORES_ON_PANIC_SGI,15)
 $(call force,CFG_INIT_CNTVOFF,y)
 $(call force,CFG_SECURE_TIME_SOURCE_CNTPCT,y)
 $(call force,CFG_WITH_ARM_TRUSTED_FW,y)
@@ -39,6 +40,7 @@ CFG_TZDRAM_SIZE  ?= 0x02000000
 CFG_CORE_HEAP_SIZE ?= 262144
 CFG_CORE_RESERVED_SHM ?= n
 CFG_DTB_MAX_SIZE ?= 262144
+CFG_HALT_CORES_ON_PANIC ?= y
 CFG_MMAP_REGIONS ?= 30
 CFG_NUM_THREADS ?= 5
 CFG_TEE_CORE_NB_CORE ?= 2
-- 
2.47.2

